docs/en/reference/working-with-associations.rst

```diff
@@ -32,62 +32,66 @@ information about its type and if it's the owning or inverse side.
 .. code-block:: php
 
     <?php
-    /** @Entity */
+    #[Entity]
     class User
     {
-        /** @Id @GeneratedValue @Column(type="string") */
-        private $id;
+        #[Id]
+        #[GeneratedValue]
+        #[Column(type: 'string')]
+        private string $id;
```

Comment:
Not sure why `string` was used here in the first place (instead of int).

Reply:
String looks wrong indeed. This should be an integer.

```suggestion
        private int|null $id = null;
```

`null` is explicitly a valid state for the entity (≠ the table) because the ORM will generate the ID on flush.

Reply:
I don't know, unitialized (no state yet) sound valid too. And I don't know in what scenario it would make sense to call `getId()` before flush.

Reply:
> I don't know, unitialized (no state yet) sound valid too.

The uninitialized state is a booby trap. Only use it if you're certain that you will never access the property before initialization. If however you need to deal with that case that the property does not have a value (yet), better make it explicit and allow `null`. This also helps static analysis tools to help you.

> And I don't know in what scenario it would make sense to call `getId()` before flush.

A possible scenario would be a piece of code that needs to check if a given entity has been persisted yet.

Reply:
Since there is no clean way to check that a property is initialized, I think you are right. People that prefer getting a clear exception can perform a null check in `getId()`.